File: Benchmark/IBench.cs
using Revenj.DomainPatterns;
using System.Collections.Generic;
using System.Linq;

namespace Benchmark
{
	public interface IBench<T>
		where T : IAggregateRoot
	{
		void Clean();
		void Analyze();
		IEnumerable<T> SearchAll();
		IEnumerable<T> SearchSubset(int i);
		IQueryable<T> Query();
		T FindSingle(string id);
		IEnumerable<T> FindMany(string[] ids);
		void Insert(IEnumerable<T> values);
		void Update(IEnumerable<T> values);
		void Insert(T value);
		void Update(T value);
		Report<T> Report(int i);
	}

	public class Report<T>
	{
		public T findOne;
		public IEnumerable<T> findMany;
		public T findFirst;
		public T findLast;
		public IEnumerable<T> topFive;
		public IEnumerable<T> lastTen;
	}
}

File: Benchmark/src/hr/ngs/benchmark/IBench.java
package hr.ngs.benchmark;

import java.util.List;

public interface IBench<T extends IAggregateRoot> {
	void clean();
	void analyze();
	List<T> searchAll();
	List<T> searchSubset(int i);
	//IQueryable<T> query();
	T findSingle(String id);
	List<T> findMany(String[] ids);
	void insert(Iterable<T> values);
	void update(Iterable<T> values);
	void insert(T value);
	void update(T value);
	Report<T> report(int i);
}
